File: tests/integration/benchmarks/fixtures/java/rosetta/src/com/rosetta/models/Report.java
package com.rosetta.models;

import java.time.Instant;
import java.util.List;

/**
 * Summary report produced by the processor.
 */
public class Report {
    private final double total;
    private final double average;
    private final List<Record> records;
    private final Instant generatedAt;

    public Report(double total, double average, List<Record> records) {
        this.total = total;
        this.average = average;
        this.records = records;
        this.generatedAt = Instant.now();
    }

    public double getTotal() { return total; }
    public double getAverage() { return average; }
    public List<Record> getRecords() { return records; }
    public Instant getGeneratedAt() { return generatedAt; }
}

File: tests/integration/benchmarks/fixtures/csharp/rosetta/src/Models/Report.cs
namespace Rosetta.Models;

/// <summary>
/// Summary report produced by the processor.
/// </summary>
public class Report
{
    public double Total { get; }
    public double Average { get; }
    public IReadOnlyList<Record> Records { get; }
    public DateTime GeneratedAt { get; }

    public Report(double total, double average, IEnumerable<Record> records)
    {
        Total = total;
        Average = average;
        Records = records.ToList().AsReadOnly();
        GeneratedAt = DateTime.UtcNow;
    }
}
